jenkins/docker/Jenkinsfile

```diff
@@ -41,25 +53,77 @@ pipeline {
             steps {
                 script {
                     git url: "$DOCKER_BUILD_GIT_REPOSITORY", branch: "$DOCKER_BUILD_GIT_REPOSITORY_REFERENCE"
-                    def CREDENTIAL_ID = "jenkins-staging-docker-staging-credential"
                     if (env.IS_STAGING == "false") {
-                        CREDENTIAL_ID = "jenkins-staging-docker-prod-token"
+                        if (!params.SOURCE_IMAGE_IN_STAGING.contains(":") || !params.DESTINATION_IMAGE_IN_PROD.contains(":")) {
+                            currentBuild.result = 'ABORTED'
+                            error('SOURCE_IMAGE_IN_STAGING parameter or DESTINATION_IMAGE_IN_PROD is incorrect')
+                        }
+                        def CREDENTIAL_ID = "jenkins-staging-docker-prod-token"
                         sh "echo Switch to Production"
+                        withCredentials([usernamePassword(credentialsId: CREDENTIAL_ID, usernameVariable: 'DOCKER_USERNAME', passwordVariable: 'DOCKER_PASSWORD')]) {
+                            sh '''
+                                set -ex
+                                echo Login to $CREDENTIAL_ID
+                                curl -L https://github.com/google/go-containerregistry/releases/latest/download/go-containerregistry_Linux_x86_64.tar.gz \\
+                                    -o go-containerregistry.tar.gz
+                                tar -zxvf go-containerregistry.tar.gz
+                                chmod +x gcrane
+                                mv gcrane /usr/local/bin/
+                                docker logout && docker login -u $DOCKER_USERNAME -p $DOCKER_PASSWORD
+                                gcrane cp ${SOURCE_IMAGE_IN_STAGING} ${DESTINATION_IMAGE_IN_PROD}
+                            '''
+                        }
+                    } else {
+                        def CREDENTIAL_ID = "jenkins-staging-docker-staging-credential"
+                        withCredentials([usernamePassword(credentialsId: CREDENTIAL_ID, usernameVariable: 'DOCKER_USERNAME', passwordVariable: 'DOCKER_PASSWORD')]) {
+                            sh '''
+                                set -ex
+                                echo Login to $CREDENTIAL_ID
```

Comment:
We should never manually log details about credentials object being used.  I believe Jenkins has logging that includes properly masked credentials info, if we need additional logging can you see what the gaps are so we can fix the plugins or wrap them properly.

Reply:
This was a part of `docker-build` job. But I can remove it from this